fullstop-jobs/src/main/java/org/zalando/stups/fullstop/jobs/ami/FetchAmiJob.java

```diff
@@ -0,0 +1,203 @@
+                                        + " " + maxValidityTimeForAmi.year().getAsText());
+                            }
+
+                            if (metaData.size() > 0) {
+                                metaData.put("errorMessages", errorMessages);
+                                writeViolation(account, region, metaData, instance.getInstanceId());
+                            }
+
+                        }
+
+                    }
+
+                } catch (AmazonServiceException a) {
+
+                    if (a.getErrorCode().equals("RequestLimitExceeded")) {
+                        log.warn("RequestLimitExceeded for account: {}", account);
+                    } else {
+                        log.error(a.getMessage(), a);
+                    }
+
+                }
+            }
+        }
+    }
+
+    private boolean isTaupageTooOld(Optional<String> imageName, DateTime now) {
+        DateTime maxValidityTimeForAmi = now.minus(Days.days(60));
+
+        String rawDate = imageName.map(s -> Stream.of(s.split("-")).collect(Collectors.toList())).get().get(2);
+
+        DateTimeFormatter formatter = DateTimeFormat.forPattern("yyyyMMdd");
+        DateTime taupageImageDate = formatter.parseDateTime(rawDate);
+
+        return taupageImageDate.isBefore(maxValidityTimeForAmi);
+    }
+
+    private void writeViolation(String account, String region, Object metaInfo, String instanceId) {
+        ViolationBuilder violationBuilder = new ViolationBuilder();
+        Violation violation = violationBuilder.withAccountId(account)
+                .withRegion(region)
+                .withPluginFullyQualifiedClassName(FetchAmiJob.class)
+                .withType(OUTDATED_AMI)
+                .withMetaInfo(metaInfo)
+                .withInstanceId(instanceId)
+                .withEventId(EVENT_ID).build();
+        violationSink.put(violation);
+    }
+
+    private DescribeInstancesResult getDescribeEC2Result(String account, String region) {
+        AmazonEC2Client ec2Client = clientProvider.getClient(
+                AmazonEC2Client.class,
+                account,
+                getRegion(
+                        fromName(region)));
+        DescribeInstancesRequest describeInstancesRequest = new DescribeInstancesRequest();
+        return ec2Client.describeInstances(describeInstancesRequest);
+    }
+
+    private Optional<List<Image>> getAmiFromEC2Api(String account, String region, final String imageId) {
+        try {
+            AmazonEC2Client ec2Client = clientProvider.getClient(
+                    AmazonEC2Client.class,
+                    account,
+                    getRegion(
+                            fromName(region)));
+
+            DescribeImagesRequest describeImagesRequest = new DescribeImagesRequest();
+            describeImagesRequest.setImageIds(newArrayList(imageId));
+            return ofNullable(ec2Client.describeImages(describeImagesRequest).getImages().stream().collect(Collectors.toList()));
```

Comment:
`getImages()` already returns a List. There is no need to stream it and then collect it again. Consider something like this: `Optional.of(ec2Client.describeImages(describeImagesRequest)).map(DescribeImagesResult::getImages)`